browser/brave_shields/brave_shields_tab_helper.cc

```diff
@@ -71,6 +82,72 @@ void BraveShieldsTabHelper::DidFinishNavigation(
       webcompat_features_invoked_.clear();
     }
     ClearAllResourcesList();
+
+    if (!navigation_triggered_by_shields_changes_) {
+      MaybeNotifyRepeatedReloads(navigation_handle);
+    }
+
+    navigation_triggered_by_shields_changes_ = false;
+  }
+}
+
+void BraveShieldsTabHelper::MaybeNotifyRepeatedReloads(
```

Comment:
This is not appropriate functionality to add to this tab helper, it's purpose is to handle shields setting changes, not to handle prompts to enable adblock only mode

Reply:
Filed an issue to fix: https://github.com/brave/brave-browser/issues/50413